cpp/src/io/avro/reader_impl.cu

```diff
@@ -0,0 +1,502 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @file reader_impl.cu
+ * @brief cuDF-IO Avro reader class implementation
+ **/
+
+#include "reader_impl.hpp"
+
+#include <io/comp/gpuinflate.h>
+#include <utilities/integer_utils.hpp>
+
+#include <cudf/table/table.hpp>
+#include <cudf/utilities/error.hpp>
+#include <cudf/utilities/traits.hpp>
+
+#include <rmm/thrust_rmm_allocator.h>
+#include <rmm/device_buffer.hpp>
+
+namespace cudf {
+namespace experimental {
+namespace io {
+namespace detail {
+namespace avro {
+
+// Import functionality that's independent of legacy code
+using namespace cudf::io::avro;
+using namespace cudf::io;
+
+namespace {
+
+/**
+ * @brief Function that translates Avro data kind to cuDF type enum
+ **/
+type_id to_type_id(const avro::schema_entry *col) {
+  switch (col->kind) {
+    case avro::type_boolean:
+      return type_id::BOOL8;
+    case avro::type_int:
+      return type_id::INT32;
+    case avro::type_long:
+      return type_id::INT64;
+    case avro::type_float:
+      return type_id::FLOAT32;
+    case avro::type_double:
+      return type_id::FLOAT64;
+    case avro::type_bytes:
+    case avro::type_string:
+      return type_id::STRING;
+    case avro::type_enum:
+      return (!col->symbols.empty()) ? type_id::STRING : type_id::INT32;
+    default:
+      return type_id::EMPTY;
+  }
+}
+
+/**
+ * @brief Function that returns the size aligned to the 4-byte multiple
+ **/
+size_t align_size(size_t size_to_align) {
```

Comment:
RMM will do this implicitly now. Is this still needed?

Reply:
Removed, thanks!